commit e1a590badee13b862c5393f45ea59ecad98a7267
Author: Eike Hein <hein@kde.org>
Date:   Mon Sep 2 20:08:08 2013 +0200

    Don't propagate drops to the containment while animating.
    
    Plus a little cleanup in the drag move and leave code ...
    
    BUG:324391

diff --git a/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml b/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml
index c2a5cdf..648e3b9 100644
--- a/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml
+++ b/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml
@@ -35,29 +35,30 @@ Item {
 
         property variant hoveredItem
 
-        enabled: !target.animating
-
         onDragMove: {
-            var above = target.childAt(event.x, event.y);
+            if (target.animating) {
+                return;
+            }
 
-            if (tasks.dragSource) {
-                if (above != tasks.dragSource
-                    && !(tasks.dragSource.isLauncher || above.isLauncher)) {
-                    var targetIndex = TaskTools.insertionIndexAt(event.x, event.y);
+            var above = target.childAt(event.x, event.y);
 
-                    itemMove(tasks.dragSource.itemId, targetIndex);
+            if (above) {
+                if (tasks.dragSource) {
+                    if (tasks.dragSource != above && !tasks.dragSource.isLauncher && !above.isLauncher) {
+                        itemMove(tasks.dragSource.itemId, TaskTools.insertionIndexAt(event.x, event.y));
+                    }
+                } else if (hoveredItem != above) {
+                    hoveredItem = above;
+                    activationTimer.start();
                 }
-            } else if (above && hoveredItem != above) {
-                hoveredItem = above;
-                activationTimer.start();
-            } else if (!above) {
+            } else {
                 hoveredItem = 0;
                 activationTimer.stop();
             }
         }
 
         onDragLeave: {
-            dragItem = 0;
+            hoveredItem = 0;
             activationTimer.stop();
         }
 
